pkg/sdk/warehouse.go

```diff
@@ -135,81 +134,99 @@ type WarehouseAlterOptions struct {
-	QueryAccelerationMaxScaleFactorField WarehouseUnsetField = "QUERY_ACCELERATION_MAX_SCALE_FACTOR"
-	MaxConcurrencyLevelField             WarehouseUnsetField = "MAX_CONCURRENCY_LEVEL"
-	StatementQueuedTimeoutInSecondsField WarehouseUnsetField = "STATEMENT_QUEUED_TIMEOUT_IN_SECONDS"
-	StatementTimeoutInSecondsField       WarehouseUnsetField = "STATEMENT_TIMEOUT_IN_SECONDS"
-)
+type WarehouseUnset struct {
+	// Object properties
+	WarehouseType                   *bool `ddl:"keyword" db:"WAREHOUSE_TYPE"`
+	WarehouseSize                   *bool `ddl:"keyword" db:"WAREHOUSE_SIZE"`
+	WaitForCompletion               *bool `ddl:"keyword" db:"WAIT_FOR_COMPLETION"`
+	MaxClusterCount                 *bool `ddl:"keyword" db:"MAX_CLUSTER_COUNT"`
+	MinClusterCount                 *bool `ddl:"keyword" db:"MIN_CLUSTER_COUNT"`
+	ScalingPolicy                   *bool `ddl:"keyword" db:"SCALING_POLICY"`
+	AutoSuspend                     *bool `ddl:"keyword" db:"AUTO_SUSPEND"`
+	AutoResume                      *bool `ddl:"keyword" db:"AUTO_RESUME"`
+	ResourceMonitor                 *bool `ddl:"keyword" db:"RESOURCE_MONITOR"`
+	Comment                         *bool `ddl:"keyword" db:"COMMENT"`
+	EnableQueryAcceleration         *bool `ddl:"keyword" db:"ENABLE_QUERY_ACCELERATION"`
+	QueryAccelerationMaxScaleFactor *bool `ddl:"keyword" db:"QUERY_ACCELERATION_MAX_SCALE_FACTOR"`
+
+	// Object params
+	MaxConcurrencyLevel             *bool              `ddl:"keyword" db:"MAX_CONCURRENCY_LEVEL"`
+	StatementQueuedTimeoutInSeconds *bool              `ddl:"keyword" db:"STATEMENT_QUEUED_TIMEOUT_IN_SECONDS"`
+	StatementTimeoutInSeconds       *bool              `ddl:"keyword" db:"STATEMENT_TIMEOUT_IN_SECONDS"`
+	Tag                             []ObjectIdentifier `ddl:"keyword" db:"TAG"`
+}
 
 func (opts *WarehouseAlterOptions) validate() error {
 	if opts.name.FullyQualifiedName() == "" {
 		return fmt.Errorf("name must not be empty")
 	}
 
-	exclusivePointers := []interface{}{
+	if err := exactlyOneValueSet(
 		opts.Suspend,
 		opts.Resume,
 		opts.AbortAllQueries,
 		opts.NewName,
 		opts.Set,
-		opts.Unset,
-		opts.SetTags,
-		opts.UnsetTags,
+		opts.Unset); err != nil {
+		return fmt.Errorf("exactly one of Suspend, Resume, AbortAllQueries, NewName, Set, Unset must be set")
 	}
-	if err := checkExclusivePointers(exclusivePointers); err != nil {
-		return fmt.Errorf("exactly one of Suspend, Resume, AbortAllQueries, NewName, Set, Unset, SetTags and UnsetTags must be set: %w", err)
-	}
-
-	if opts.Suspend != nil && *opts.Suspend && opts.Resume != nil && *opts.Resume {
-		return fmt.Errorf(`"Suspend" and "Resume" cannot both be true`)
+	if everyValueSet(opts.Suspend, opts.Resume) && (*opts.Suspend && *opts.Resume) {
+		return fmt.Errorf("Suspend and Resume cannot both be true")
 	}
-	if opts.IfSuspended != nil && *opts.IfSuspended && (opts.Resume == nil || !*opts.Resume) {
+	if valueSet(opts.IfSuspended) && !valueSet(opts.Resume) {
```

Comment:
We also need to check the values of IfSuspended and Resume, not just their values:
```suggestion
	if valueSet(opts.IfSuspended) && *opts.ifSuspended && (!valueSet(opts.Resume) || !*opts.Resume) {
```